src/libmongoc/src/mongoc/mongoc-trace-private.h

```diff
@@ -33,41 +33,41 @@ BSON_BEGIN_DECLS
 
```

Comment:
> Mild concern that moving the MONGOC_TRACE_ENABLED check into the function may impede the compiler's ability to deduce always-false conditions at usage sites.

Given that `ENTRY` and `EXIT` are called in many functions, I would rather err towards avoiding the extra runtime cost.

> this may be premature optimization

Agreed, but I expect it is a low complexity optimization. I suggest: remove `static` from `gLogTrace` in mongoc-log.c to enable external linkage and use as follows:

```c
// `gLogTrace` determines if tracing is enabled at runtime.
extern bool gLogTrace;

#define TRACE(msg, ...)                                                                                          \
   do {                                                                                                          \
      if (MONGOC_TRACE_ENABLED && gLogTrace) {                                                                   \
         mongoc_log (                                                                                            \
            MONGOC_LOG_LEVEL_TRACE, MONGOC_LOG_DOMAIN, "TRACE: %s():%d " msg, BSON_FUNC, __LINE__, __VA_ARGS__); \
      }                                                                                                          \
   } while (0)
```